### DISCUSSION TITLE
What is the correct example to use XOAUTH2 with PHPmailer with GMAIL after obtaining TOKEN?

### DISCUSSION THREAD

**Discussion by zagarskas** [PHPMailer/PHPMailer#2891] [Q&A]
https://github.com/PHPMailer/PHPMailer/blob/master/examples/gmail_xoauth.phps
Is this... up to date?

_For example, I have gone through this setup:_
https://github.com/PHPMailer/PHPMailer/wiki/Using-Gmail-with-XOAUTH2

- Composer is all good (I can use regular PHPmailer on old SMTP method)
- got all the XOAUTH2 keys, setup, everything looks good
- but the documentation here: /PHPMailer/blob/master/examples/gmail_xoauth.phps 
- seems, "not correct" when attempted in context of the Using-Gmail-with-XOAUTH2 **article**

after following `Using-Gmail-with-XOAUTH2` then `gmail_xoauth.phps` does not work
Not sure what I am missing here, but I think the explanation on Using-Gmail-with-XOAUTH2 is confusing, even so, I should be able to drop in my token and get a result... maybe I am mixing up the settings.

Which of the following conclusions are true here:

- The example given on `Using-Gmail-with-XOAUTH2` does not show that `$mail->Host` and SMTP is still required, but it is. I have therefore misinterpreted the XOAUTH2 example to mean SMTP is not required, which is why XOAUTH2 is still not a solution for a host like GoDaddy
- The example given on `Using-Gmail-with-XOAUTH2` does not require any SMTP auth and `$mail->isSMTP();` is not required, instead `$mail->oauthRefreshToken` is used which allows one to bypass SMTP sending, making it a good replacement for SMTP troubles on hosts like GoDaddy. However, I am simply implementing it wrong.

TLDR version:

- Yes: SMTP auth is still required when using XOAUTH2 via `Using-Gmail-with-XOAUTH2`
- No: SMTP auth is not required when using XOAUTH2 via `Using-Gmail-with-XOAUTH2`
